docs/sdk/web3-react/guides/01-connect-wallet.md

```diff
@@ -1,4 +1,165 @@
+
+The return type of the function is a tuple of the initialized **connector** and the **hooks** that we can use on it. Using this tuple, we create an instance of a [`Connection`](https://github.com/Uniswap/examples/blob/856dbb002e7f38120554ef226f4309c96ce6ea79/web3-react/src/libs/connections.ts#L10) type, by setting the `type` attribute to `INJECTED`:
+
+
+```typescript reference title="Creating a connection instance" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/856dbb002e7f38120554ef226f4309c96ce6ea79/web3-react/src/libs/injected.ts#L16-L20
+```
+
+Finally, we return the instance we created, which is added to the list of prioritized connectors. 
+
+:::info
+For help on creating the rest of the supported connectors of this examples, please visit our [connectors](./connectors.md) page!
+:::
+
+
+## Connecting and  disconnecting the application to the connector
+
+Having built our Injected Connector, we can now build the component that gives our application the ability to communicate with that connector. The basic functionality we would like to support is **connecting** and **disconnecting**  the connector to our application. To achieve this, we create a reusable [Option component](https://github.com/Uniswap/examples/blob/main/web3-react/src/libs/components/Option.tsx) that supports rendering the UI and managing state for any connector that we want to support in our application:
+
+```typescript reference title="Creating the Option component" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/81ec93e97b0afded621e177fe5f34fc9f98f80b0/web3-react/src/libs/components/Option.tsx#L5-L11
+```
+
+The component receives 5 parameters:
+- whether the connector is eligible for to be activated
+- whether the connector is currently active 
+- the `ConnectionType`
+- the last 2 are hooks which are called once the component has established a connection or has been disconnected. 
+
+In the case of *MetaMask*, when declaring the `InjectedConnector` we pass the connector specific arguments:
+
+```typescript reference title="Creating an injected connector" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/81ec93e97b0afded621e177fe5f34fc9f98f80b0/web3-react/src/libs/components/ConnectionOptions.tsx#L26-L32
+```
+
+It is worth noting that there are 2 ways for `isEnabled` to be `true`. Either the option is the currently active option, and so the `Option` is enabled so we can disconnect. The other case is when there is no `Option` active: 
```

Comment:
I don't understand this.

Reply:
Doesn't really add much - better to omit